centreon/www/include/configuration/configObject/service_categories/DB-Func.php

```diff
@@ -156,64 +167,141 @@ function multipleServiceCategorieInDB($sc = [], $nbrDup = [])
                 $statement = $pearDB->query("SELECT MAX(sc_id) as maxid FROM `service_categories`");
                 $maxId = $statement->fetch();
-                $scAcl[$maxId['maxid']] = $key;
-                $query = "INSERT INTO service_categories_relation (service_service_id, sc_id) " .
-                    "(SELECT service_service_id, :max_id" .
-                    " FROM service_categories_relation WHERE sc_id = :sc_id)";
-                $statement = $pearDB->prepare($query);
-                $statement->bindValue(':sc_id', $key, \PDO::PARAM_INT);
-                $statement->bindValue(':max_id', $maxId['maxid'], \PDO::PARAM_INT);
-                $statement->execute();
+
+                if (isset($maxId['maxid'])) {
+                    $scAcl[$maxId['maxid']] = $scId;
+                    $selectServiceIdsStatement = $pearDB->prepare(
+                        <<<SQL
+                            SELECT service_service_id FROM service_categories_relation
+                            WHERE sc_id = :sc_id
+                        SQL
+                    );
+                    $selectServiceIdsStatement->bindValue(':sc_id', $scId, \PDO::PARAM_INT);
+                    $selectServiceIdsStatement->execute();
+                    $insertNewRelationStatement = $pearDB->prepare(
+                        <<<SQL
+                            INSERT INTO service_categories_relation (service_service_id, sc_id)
+                            VALUES (:serviceId, :maxId)
+                        SQL
+                    );
+                    $foundServiceIds = [];
+                    while ($serviceId = $selectServiceIdsStatement->fetchColumn()) {
+                        $insertNewRelationStatement->bindValue(':serviceId', $serviceId, \PDO::PARAM_INT);
+                        $insertNewRelationStatement->bindValue(':maxId', $maxId['maxid'], \PDO::PARAM_INT);
+                        $insertNewRelationStatement->execute();
+                        $foundServiceIds[] = $serviceId;
+                    }
+                    if (! empty($foundServiceIds)) {
+                        $fields["sc_services"] = implode(", ", $foundServiceIds);
+                    }
+
+                    $centreon->CentreonLogAction->insertLog(
+                        object_type: ActionLog::OBJECT_TYPE_SERVICECATEGORIES,
+                        object_id: $maxId['maxid'],
+                        object_name: $sc_name,
+                        action_type: ActionLog::ACTION_TYPE_ADD,
+                        fields: $fields
+                    );
+                }
             }
         }
     }
     CentreonACL::duplicateScAcl($scAcl);
     $centreon->user->access->updateACL();
 }
 
-function enableServiceCategorieInDB(?int $sc_id = null, $sc_arr = [])
+function enableServiceCategorieInDB(?int $serviceCategoryId = null, $serviceCategories = [])
 {
-    if (!$sc_id && !count($sc_arr)) {
+
+    if (! $serviceCategoryId && ! count($serviceCategories)) {
```

Comment:
Empty is fatser than count :) 
```suggestion
    if (! $serviceCategoryId && empty($serviceCategories)) {
```